python/snewpy/flux.py

```diff
@@ -361,9 +386,29 @@ def __eq__(self, other:'Container')->bool:
         result = self.__class__==other.__class__ and \
                  self.unit == other.unit and \
                  np.allclose(self.array, other.array) and \
-                 all([np.allclose(self.axes[ax], other.axes[ax]) for ax in Axes])
+                 self.flavor_scheme==other.flavor_scheme and \
+                 len(self.flavor)==len(other.flavor) and \
+                 all(self.flavor==other.flavor) and \
+                 all([np.allclose(self.axes[ax], other.axes[ax]) for ax in list(Axes)[1:]])
         return result
 
+    def _is_full_flavor(self):
+        return all(self.flavor==list(self.flavor_scheme))
+                   
+    def convert_to_flavor(self, flavor:FlavorScheme):
+        if(self.flavor_scheme==flavor):
+            return self
+        return (self.flavor_scheme>>flavor)@self
+    def __rshift__(self, flavor:FlavorScheme):
+        return self.convert_to_flavor(flavor)
```

Comment:
The `>>` and `<<` operators took me a moment to wrap my head around. Now that I’m used to them, I find them fairly elegant. The awkward thing about them is the difference between these two lines:
```Python
E_GeV = get_energies(…) << u.GeV
flux_3f = flux >> ThreeFlavor
```
Conceptually these feel very similar to me (I enforce a particular representation onto a physical quantity), but in one case I use `<<` and in the other I use `>>`.
Would you be open to turning this into a `__lshift__`, to be consistent with astropy.units?

Reply:
The operators look nice in the matrix definitions - the arrows show the conversion direction:
```python
M = (TwoFlavor>>ThreeFlavor) #converts 2f to 3f
M = (ThreeFlavor<<TwoFlavor) #same, but just different operator
M = (FourFlavor<<ThreeFlavor) @ (ThreeFlavor<<TwoFlavor) #looks rather natural to me
```
So when we have 
```python
flux>>ThreeFlavor
```
it is also the same: we expect that final flavor is the ThreeFlavor, whereas
```python
flux<<ThreeFlavor
```
looks strange (to me).

It makes more sense for `astropy.units` because you *assign* the unit to the quantity:
```python
E_GeV = get_energies(…) << u.GeV #get_energies could be just np.array with no unit, so we attach u.GeV to it
```
so I don't think we necessarily need to make these cases similar

Reply:
`FourFlavor >> ThreeFlavor` is a really nice and intuitive way of creating a transformation matrix, I agree.

But if both sides have different types, my intuition is that the resulting object is a modified version of the object that the operator points to. So I would read these cases as follows:
```Python
some_quantity << u.GeV  # results in a quantity, with a particular unit attached
flux << ThreeFlavor  # results in a flux, with a particular flavour scheme enforced
flux >> ThreeFlavor  # results in a ThreeFlavor scheme with …?!?
```

To me, the analogy to the `astropy.units` example is stronger than the analogy to the `FourFlavor >> ThreeFlavor` case; but I see that either is reasonable; I’d be interested in hearing from others what their gut feeling is. But if there’s no clear preferences for either option, I’m fine with leaving it the way it is now; I’m sure I’ll get used to it.

Reply:
I think we resolved this, keeping both >> and << for now.